api/middlewares/user_middleware.js

```diff
@@ -0,0 +1,85 @@
+const Users = require('./../models/user_model');
+const mongoose = require('mongoose');
+const jwt = require('jsonwebtoken');
+
+const validateUserId = (req, res, next) => {
+    if(mongoose.Types.ObjectId.isValid(req.params.userId)) {
+        Users.getById(req.params.userId).then((user) => {
+            if(!user) {
+                return res.status(404).json({
+                    msg: 'No user found'
+                })
+            }
+            else
+            next();
+        }).catch((err) => {
+            next(err);
+        })
+        
+    }
+    else
+    return res.status(404).json({msg: 'id not valid'})
+}
+
+const validateUser = (req, res, next) => {
+    if(!req.body.username || !req.body.password) {
+        return res.status(400).json({
+            msg: 'Invalid credentials'
+        })
+    }
+    else
+    Users.getAll().then((users) => {
```

Comment:
de ce sa nu faci cautarea in baza de date dupa username. Imagineaza-ti ca ai avea in baza 100000 de utilizatori, acest cod ar fi fost imposibil de executat

Reply:
Am corectat.